docs/enhancements/ai-on-zero-worker-hypershift.md

```diff
@@ -0,0 +1,118 @@
+2. The operator reads the current CRDs on the management system. Once the CRDs are known to the operator, it will use a runtime client pointed to the L1 api-server to register the CRDs
+
+#### Client Configuration
+In the assisted service, we define a KUBECONFIG env variable to the service deployment that points to a mounted secret containing L1's kubeconfig. This secret is defined in a well-defined place in the management cluster and is also used by ACM. In this way, the deployed service to namespace <b>k</b> communicates with its corresponding hypershift hosted cluster out of the box with no code changes required.
+
+
+Open Issue: How and where is that secret provisioned and become known to us (AI/ACM integration issue)
+
+### HypershiftServerConfig controller (Infrastructure operator)
+#### Client configuration
+EXTERNAL_KUBECONFIG  env variable to the service deployment points to a mounted secret containing L1's kubeconfig. This secret is defined in a well-defined place in the management cluster and is also used by ACM.
+
+The controller shall register itself to watch and manage resources with both L0 and L1 api-servers and instantiate 2 clients, one for each control plane.
+
+#### Separating L0 and L1 resources
+The resources to watch will be created by a factory and divided into L0 and L1 resources. 
+
+The infrastructure controller then uses the appropriate clients and the inventory of resources to provision and update the necessary configuration according to the scheme above.
+
+#### RBAC
+As with the issue of CRDs, the operator needs to define the RBAC per L1 deployment. The same options for finding which resources we need to deploy:
+1. Build mechanism takes the RBAC from the deploy yaml and stores them on the filesystem of the operator
+2. The operator reads the current RBAC definition on the management system
+
+Option 1 is the selected one.
+
+### Webhooks
+Reference: [Dynamic Admission Control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/)
+Admission webhooks are HTTP callbacks that receive admission requests and do something with them. You can define two types of admission webhooks, validating admission webhook and mutating admission webhook. To register admission webhooks, create MutatingWebhookConfiguration or ValidatingWebhookConfiguration API objects.
+These configuration objects need to be registered with L1 api-server.
+
+The admission logic is served by an admission webhook server which serves as an extension to the api server. The pod that host the webhook server is deployed on the management cluster (L0).
+
+In our current assisted-service deployment the communication between the api-server and the webhook server is done by a service since both runs on the same cluster. In the hypershift case, it should be done via konnectivity agent and client.
+
+To accomplish that we need to:
```

Comment:
Could we just expose the webhook with a Route on L0, then on L1 refer to that route's URL when registering the webhook?

Reply:
I think there is an egress configuration in L1 kube-apiserver based on the design of hypershift. All traffic of L1 kube-apiserver will go to konnectivity-server.

And konnectivity-server only can proxy traffic to konnectivity-agent.  So a Route may not help in this case.